### PR TITLE
Migrate Microsoft.Bot.Builder.Dialogs.Tests to xUnit

### PR DESCRIPTION
**Pull request by sw-joelmut**
Fixes # 4096

## Description
This PR migrates **the following test files** inside the [Microsoft.Bot.Builder.Dialogs.Tests](https://github.com/microsoft/botbuilder-dotnet/tree/main/tests/Microsoft.Bot.Builder.Dialogs.Tests) folder from **MSTest** to **xUnit**.
- ActivityPromptTests.cs
- AttachmentPromptTests.cs
- ChoiceFactoryTests.cs
- ChoicePromptTests.cs
- ChoicesChannelTests.cs
- ChoicesRecognizersTests.cs
- ChoicesTokenizerTests.cs

## Specific Changes
- Added `xUnit` packages to `.csproj`.
- Changed `[TestMethod]` to `[Fact]`.
- Changed `[TestCategory]` to `[Trait]`.
- Changed `[ExpectedException]` to `Assert.Throws`.
- Changed `Assert.AreEqual` to `Assert.Equal`.
- Changed `Assert.IsNotNull` to `Assert.NotNull`.
- Changed `Assert.IsNull` to `Assert.Null`.
- Changed `Assert.IsTrue` to `Assert.True`.
- Changed ` Assert.IsInstanceOfType` to `Assert.IsAssignableFrom`.
- Improved tests Assertions.

## Testing
In the following image there are the passing tests.
image

### CODE REVIEW

**Review by iscaltritti-sw — APPROVED** [review]

File: tests/Microsoft.Bot.Builder.Dialogs.Tests/ChoicePromptTests.cs [hunk 1]
```diff
@@ -27,21 +25,53 @@ public class ChoicePromptTests
             new Choice { Value = "blue" },
         };
 
-        [TestMethod]
-        [ExpectedException(typeof(ArgumentNullException))]
+        /// <summary>
+        /// Generates an Enumerable of variations on all supported locales.
+        /// </summary>
+        /// <returns>An iterable collection of objects.</returns>
+        public static IEnumerable<object[]> GetLocaleVariationTest()
```
Comment (ceciliaavila) [inline tests/Microsoft.Bot.Builder.Dialogs.Tests/ChoicePromptTests.cs:32]: Why did you change this method from private to public?

File: tests/Microsoft.Bot.Builder.Dialogs.Tests/ChoicePromptTests.cs [hunk 2]
```diff
@@ -820,25 +850,25 @@ private Action<IActivity> StartsWithValidator(string expected)
         {
             return activity =>
             {
-                Assert.IsInstanceOfType(activity, typeof(IMessageActivity));
+                Assert.IsAssignableFrom<IMessageActivity>(activity);
```
Comment (ceciliaavila) [inline tests/Microsoft.Bot.Builder.Dialogs.Tests/ChoicePromptTests.cs:853]: Shouldn't we use `IsType<T>`?
```suggestion
                Assert.IsType<IMessageActivity>(activity);
```

File: tests/Microsoft.Bot.Builder.Dialogs.Tests/ChoicePromptTests.cs [hunk 3]
```diff
@@ -820,25 +850,25 @@ private Action<IActivity> StartsWithValidator(string expected)
...
+                Assert.StartsWith(expected, msg.Text);
             };
         }
 
         private Action<IActivity> SuggestedActionsValidator(string expectedText, SuggestedActions expectedSuggestedActions)
         {
             return activity =>
             {
-                Assert.IsInstanceOfType(activity, typeof(IMessageActivity));
+                Assert.IsAssignableFrom<IMessageActivity>(activity);
```
Comment (ceciliaavila) [inline tests/Microsoft.Bot.Builder.Dialogs.Tests/ChoicePromptTests.cs:863]:
```suggestion
                Assert.IsType<IMessageActivity>(activity);
```

File: tests/Microsoft.Bot.Builder.Dialogs.Tests/ChoicePromptTests.cs [hunk 4]
```diff
@@ -847,18 +877,18 @@ private Action<IActivity> HeroCardValidator(HeroCard expectedHeroCard, int index
...
```
Comment (ceciliaavila) [inline tests/Microsoft.Bot.Builder.Dialogs.Tests/ChoicePromptTests.cs:880]:
```suggestion
                Assert.IsType<IMessageActivity>(activity);
```

File: tests/Microsoft.Bot.Builder.Dialogs.Tests/ChoicePromptTests.cs [hunk 5]
```diff
@@ -867,45 +897,11 @@ private Action<IActivity> SpeakValidator(string expectedText, string expectedSpe
...
```
Comment (ceciliaavila) [inline tests/Microsoft.Bot.Builder.Dialogs.Tests/ChoicePromptTests.cs:900]: Same in here.

**Review by ceciliaavila — CHANGES_REQUESTED** [review]

File: tests/Microsoft.Bot.Builder.Dialogs.Tests/ChoicePromptTests.cs (hunk 2 above)
Comment (sw-joelmut) [reply tests/Microsoft.Bot.Builder.Dialogs.Tests/ChoicePromptTests.cs:853]: It recommends in the potential fixes to use IsAssignableFrom, otherwise it shows the xUnit error.
image

Comment (ceciliaavila) [reply tests/Microsoft.Bot.Builder.Dialogs.Tests/ChoicePromptTests.cs:853]: Ok, thanks!

File: tests/Microsoft.Bot.Builder.Dialogs.Tests/ChoicePromptTests.cs (hunk 1 above)
Comment (sw-joelmut) [reply tests/Microsoft.Bot.Builder.Dialogs.Tests/ChoicePromptTests.cs:32]: It recommends using it as public otherwise it shows an xUnit error.
image

**Review by ceciliaavila — APPROVED** [review]

**Review by denscollo — APPROVED** [review]

**Comment by denscollo** [comment]
Promoted
